firedrake/functionspacedata.py

```diff
@@ -538,13 +540,28 @@ def get_map(self, V, entity_set, map_arity, bcs, name, offset, parent):
             else:
                 new_entity_node_list = entity_node_list
 
+            if kind == "interior_facet" and self.bt_masks is not None:
+                bt_masks = {}
+                off = map_arity // 2
+                for method, (bottom, top) in iteritems(self.bt_masks):
+                    b = []
+                    t = []
+                    for i in bottom:
+                        b.append(i)
+                        b.append(i+off)
+                    for i in top:
+                        t.append(i)
+                        t.append(i+off)
+                    bt_masks[method] = (b, t)
```

Comment:
after creation, we don't know we need to do the duplication.  Notice the disgusting new `kind` argument to this function.